simphony/io/tests/test_file_lattice.py

```diff
@@ -0,0 +1,180 @@
+                    N = LatticeNode((i+1, j+1, k),
+                                    {CUBA.MATERIAL_ID: 0,
+                                     CUBA.DENSITY: 1.0,
+                                     CUBA.VELOCITY: [0.0, 0.0, 1.0]})
+                    self.lattice.update_node(N)
+
+        lat = self.lattice
+        self.filelattice = FileLattice(self.file._handle, lat.name, lat.type,
+                                       lat.base_vect, lat.size, lat.origin,
+                                       CustomRecord)
+        for N in lat.iter_nodes():
+            self.filelattice.update_node(N)
+
+    def tearDown(self):
+        if os.path.exists(self.filename):
+            self.file.close()
+        shutil.rmtree(self.temp_dir)
+
+    def test_file_lattice_constructed_from_Lattice(self):
+        """ Checks that FileLattice constructed from Lattice with a
+        CustomRecord column description has correct attributes
+
+        """
+        self.assertEqual(self.filelattice.name, 'test_lattice')
+        self.assertEqual(self.filelattice.type, 'cubic')
+        self.assertItemsEqual(self.filelattice.base_vect,
+                              numpy.array((1.0, 1.0, 1.0)))
+        self.assertItemsEqual(self.filelattice.size, (4, 5, 6))
+        self.assertItemsEqual(self.filelattice.origin,
+                              numpy.array((0.0, 0.0, 0.0)))
+
+    def test_file_lattice_constructed_from_table_existing_in_file(self):
+        """ Checks that FileLattice constructed from a known table already
+        existing in H5CUDS-file provides correct attributes for FileLattice
+        object
+
+        """
+        self.filelattice = self.file.get_lattice('test_lattice')
+
+        self.assertEqual(self.filelattice.name, 'test_lattice')
+        self.assertEqual(self.filelattice.type, 'cubic')
+        self.assertItemsEqual(self.filelattice.base_vect,
+                              numpy.array((1.0, 1.0, 1.0)))
+        self.assertItemsEqual(self.filelattice.size, (4, 5, 6))
+        self.assertItemsEqual(self.filelattice.origin,
+                              numpy.array((0.0, 0.0, 0.0)))
+
+    def test_get_node(self):
+        """ Check that a LatticeNode can be retrieved correctly
+
+        """
+        N = self.filelattice.get_node((3, 3, 3))
+
+        self.assertTrue(isinstance(N, LatticeNode))
+        self.assertItemsEqual(N.index, (3, 3, 3))
+
+        for key, value in self.lattice.get_node((3, 3, 3)).data.iteritems():
+            self.assertEqual(N.data[key], value)
+
+    def test_node_iterator(self):
+        """ Checks the node iterator
+
+        Checks that an iterator over all
+        the nodes is returned when the function iter_nodes is called
+        without arguments
+
+        """
+        fl_nodes = self.filelattice.iter_nodes()
+
+        for M in fl_nodes:
+            N = self.lattice.get_node(M.index)
+            self.assertEqual(N.index, M.index)
+            self.assertItemsEqual(N.data, M.data)
```

Comment:
This will compare only the keys in the `DataContainers`. I would suggest to use the utility methods in https://github.com/simphony/simphony-common/blob/master/simphony/testing/utils.py#L26